bb8/src/inner.rs

```diff
@@ -149,12 +149,20 @@ where
             "handled in caller"
         );
 
+        let max_lifetime = self.inner.statics.max_lifetime;
+        let is_expired = max_lifetime.map_or(false, |lt| conn.is_expired(Instant::now() - lt));
+        let is_broken = self.inner.manager.has_broken(&mut conn.conn);
+
         let mut locked = self.inner.internals.lock();
-        match (state, self.inner.manager.has_broken(&mut conn.conn)) {
-            (ConnectionState::Present, false) => locked.put(conn, None, self.inner.clone()),
-            (_, is_broken) => {
+        match (state, is_broken || is_expired) {
+            (ConnectionState::Present, false) if !is_expired => {
```

Comment:
So we don't need `if !is_expired` anymore, right?

Reply:
Good catch, I totally missed that